app/controllers/laboratories_controller.rb

```diff
@@ -7,37 +7,53 @@ class LaboratoriesController < ApplicationController
   before_action :check_can_edit, only: %i[update destroy]
   before_action :check_patient
   before_action :check_lab, only: %i[update destroy]
-  rescue_from ActiveRecord::RecordInvalid, with: :handle_validation_error
 
   # Create a new lab result
   def create
     lab = Laboratory.new(lab_type: params.permit(:lab_type)[:lab_type],
                          specimen_collection: params.permit(:specimen_collection)[:specimen_collection],
                          report: params.permit(:report)[:report],
-                         result: params.permit(:result)[:result])
-    lab.patient_id = @patient_id
-    lab.save!
-    History.lab_result(patient: @patient_id,
-                       created_by: current_user.email,
-                       comment: "User added a new lab result (ID: #{lab.id}).")
+                         result: params.permit(:result)[:result],
+                         patient_id: @patient_id)
+
+    # Handle lab creation success or failure
+    if lab.save
+      # Create history item on successful create
+      History.lab_result(patient: @patient_id,
+                         created_by: current_user.email,
+                         comment: "User added a new lab result (ID: #{lab.id}).")
+    else
+      # Handle case where lab create failed
+      error_message = 'Lab result was unable to be created.'
+      render(json: { error: error_message }, status: :bad_request) && return
+    end
   end
 
   # Update an existing lab result
   def update
-    @lab.update!(lab_type: params.permit(:lab_type)[:lab_type],
-                 specimen_collection: params.permit(:specimen_collection)[:specimen_collection],
-                 report: params.permit(:report)[:report],
-                 result: params.permit(:result)[:result])
+    update_params = {
+      lab_type: params.permit(:lab_type)[:lab_type],
+      specimen_collection: params.permit(:specimen_collection)[:specimen_collection],
+      report: params.permit(:report)[:report],
+      result: params.permit(:result)[:result]
+    }
```

Comment:
Why this change? I'd prefer passing the arguments directly to the update method.

Reply:
added so line 42 (checking if the update worked) is cleaner.  We do this in the `vaccine_contoller` as well

Reply:
I'm generally not a fan of creating variables to be used once, unless readability is really bad, and I think the readability here was fine before. It's not really worth debating though, so I'm gonna just close this thread.